lib/bundler/installer/standalone.rb

```diff
@@ -25,10 +25,15 @@ def generate
 
     def paths
       @specs.map do |spec|
-        next if spec.name == "bundler"
-        Array(spec.require_paths).map do |path|
-          gem_path(path, spec).sub(version_dir, '#{ruby_engine}/#{ruby_version}')
-          # This is a static string intentionally. It's interpolated at a later time.
+        begin
+          next if spec.name == "bundler"
+          Array(spec.require_paths).map do |path|
```

Comment:
`gem_path` is where the exception is occurring, but I see your point regarding moving the exception handling down a level/method call.